ios/brave-ios/Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift

```diff
@@ -538,23 +548,51 @@ private class DisplayURLLabel: UILabel {
 
   override var font: UIFont! {
     didSet {
-      updateTextSize(from: text)
+      updateText()
+      updateTextSize()
     }
   }
 
   override var text: String? {
     didSet {
       clippingFade.isHidden = true
       if oldValue != text {
-        updateTextSize(from: text)
+        updateText()
+        updateTextSize()
         detectLanguageForNaturalDirectionClipping()
       }
       setNeedsDisplay()
     }
   }
 
-  private func updateTextSize(from value: String?) {
-    textSize = (value as? NSString)?.size(withAttributes: [.font: font!]) ?? .zero
+  private func updateText() {
+    if let text = text {
+      let attributedString = NSMutableAttributedString(string: text)
+      let paragraphStyle = NSMutableParagraphStyle()
+      paragraphStyle.lineBreakMode = .byClipping
+      paragraphStyle.baseWritingDirection = .leftToRight
+
+      if let font = font {
+        attributedString.addAttribute(
+          .font,
+          value: font,
+          range: NSRange(location: 0, length: attributedString.length)
+        )
+      }
+
+      attributedString.addAttribute(
+        .paragraphStyle,
+        value: paragraphStyle,
+        range: NSRange(location: 0, length: attributedString.length)
+      )
+      self.attributedText = attributedString
```

Comment:
Why is this all needed when we're already overriding the underlying `drawText` method to adjust the visible rect?

Reply:
The underlying draw text does not properly handle right alignment in Arabic. I tested a mixed URL consisting of English and Arabic and the underlying code will always right align which is wrong if the domain itself is English (left aligned).